### ISSUE TITLE
(src improvement) Some source code potential problem.

### ISSUE THREAD

**Issue by stevenleeS0ht** [siacs/Conversations#3535]
# 1.

eu/siacs/conversations/ui/widget/CopyTextView.java
```java
import android.widget.TextView;


public class CopyTextView extends TextView {

	public CopyTextView(Context context) {
		super(context);
	}

	public CopyTextView(Context context, AttributeSet attrs) {
		super(context, attrs);
	}

```
This custom view should extend "android.support.v7.widget.AppCompatTextView" instead.

And when I switch to `AppCompatTextView`, it seems that some super call (`super(...)`) are not implemented by AppCompatTextView. I am wondering if there is space to make improvement on this source code.

# 2

eu/siacs/conversations/services/NotificationService.java
```java
    void updateErrorNotification() {
        if (Config.SUPPRESS_ERROR_NOTIFICATION) {
            cancel(ERROR_NOTIFICATION_ID);
            return;
        }
        final boolean showAllErrors = QuickConversationsService.isConversations();
        final List<Account> errors = new ArrayList<>();
        for (final Account account : mXmppConnectionService.getAccounts()) {
            if (account.hasErrorStatus() && account.showErrorNotification() && (showAllErrors || account.getLastErrorStatus() == Account.State.UNAUTHORIZED)) {
                errors.add(account);
            }
        }
        if (mXmppConnectionService.foregroundNotificationNeedsUpdatingWhenErrorStateChanges()) {
            notify(FOREGROUND_NOTIFICATION_ID, createForegroundNotification());
        }
        final Notification.Builder mBuilder = new Notification.Builder(mXmppConnectionService);
        if (errors.size() == 0) {
            cancel(ERROR_NOTIFICATION_ID);
            return;
        } else if (errors.size() == 1) {
            mBuilder.setContentTitle(mXmppConnectionService.getString(R.string.problem_connecting_to_account));
            mBuilder.setContentText(errors.get(0).getJid().asBareJid().toString());
        } else {
            mBuilder.setContentTitle(mXmppConnectionService.getString(R.string.problem_connecting_to_accounts));
            mBuilder.setContentText(mXmppConnectionService.getString(R.string.touch_to_fix));
        }
        mBuilder.addAction(R.drawable.ic_autorenew_white_24dp,
                mXmppConnectionService.getString(R.string.try_again),
                createTryAgainIntent());
        mBuilder.setDeleteIntent(createDismissErrorIntent());
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            mBuilder.setVisibility(NotificationCompat.VISIBILITY_PRIVATE);
         // ////////////////////\\\\\\\\\\\\\\\\\\\\


            mBuilder.setSmallIcon(R.drawable.ic_warning_white_24dp);
        } else {
            mBuilder.setSmallIcon(R.drawable.ic_stat_alert_warning);
        }
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT_WATCH) {
            mBuilder.setLocalOnly(true);
        }
        mBuilder.setPriority(Notification.PRIORITY_LOW);
        final Intent intent;
        if (AccountUtils.MANAGE_ACCOUNT_ACTIVITY != null) {
            intent = new Intent(mXmppConnectionService, AccountUtils.MANAGE_ACCOUNT_ACTIVITY);
        } else {
            intent = new Intent(mXmppConnectionService, EditAccountActivity.class);
            intent.putExtra("jid", errors.get(0).getJid().asBareJid().toEscapedString());
            intent.putExtra(EditAccountActivity.EXTRA_OPENED_FROM_NOTIFICATION, true);
        }
        mBuilder.setContentIntent(PendingIntent.getActivity(mXmppConnectionService, 145, intent, PendingIntent.FLAG_UPDATE_CURRENT));
        if (Compatibility.runsTwentySix()) {
            mBuilder.setChannelId("error");
        }
        notify(ERROR_NOTIFICATION_ID, mBuilder.build());
    }

```

The line above `// ////////////////////\\\\\\\\\\\\\\\\\\\\`,  It seems should be `Notification.XXXX`(`Notification.VISIBILITY_PRIVATE`) instead of `NotificationCompat.VISIBILITY_PRIVATE`. But I am not sure.

# 3
eu/siacs/conversations/ui/ShareLocationActivity.java
```java
	protected void updateUi() {
		if (!hasLocationFeature || noAskAgain || isLocationEnabledAndAllowed()) {
			this.snackBar.dismiss();
		} else {
			this.snackBar.show();
		}

		if (isLocationEnabledAndAllowed()) {
			this.binding.fab.setVisibility(View.VISIBLE);
// ///////////////////////////////\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\
			runOnUiThread(() -> {
				this.binding.fab.setImageResource(marker_fixed_to_loc ? R.drawable.ic_gps_fixed_white_24dp :
						R.drawable.ic_gps_not_fixed_white_24dp);
				this.binding.fab.setContentDescription(getResources().getString(
						marker_fixed_to_loc ? R.string.action_unfix_from_location : R.string.action_fix_to_location
				));
				this.binding.fab.invalidate();
			});
		} else {
			this.binding.fab.setVisibility(View.GONE);
// //////////////////////////\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\
		}
	}
}
```
The lines above `// //////////////////////////\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\`.
VisibilityAwareImageButton.setVisibility can only be called from within the same library group (groupId=com.android.support)

# 4
eu/siacs/conversations/ui/ShowLocationActivity.java'

```java
	protected void updateUi() {
		final Intent i = new Intent(Intent.ACTION_VIEW, Uri.parse("google.navigation:q=0,0"));
		final ComponentName component = i.resolveActivity(getPackageManager());
		this.binding.fab.setVisibility(component == null ? View.GONE : View.VISIBLE);
	}
```
same problem.

# 5
eu/siacs/conversations/persistance/DatabaseBackend.java
```java
            + SQLiteAxolotlStore.KEY + " TEXT, FOREIGN KEY("
```
`<column definition name>` or `<table constraint>` expected, got `key`.
there is muptiple places have this problem.

# 6
eu/siacs/conversations/persistance/DatabaseBackend.java
```java
    private static String CREATE_MESSAGE_INDEX_TABLE = "CREATE VIRTUAL TABLE messages_index USING FTS4(uuid TEXT PRIMARY KEY, body TEXT)";
```
')' or comma expected, got 'TEXT'.

The above problem don't affect final build. But I am wondering if there is more formal and recommended way to implement these functions.

**Comment by stevenleeS0ht** [comment]
@iNPUTmice , do you have any opinion about the points above?

**Comment by iNPUTmice** [comment]
Two of these have been fixed in a39fff5c8efe174beb5fd72ed2816dbfeafacd73. Fixing the others isn’t that simple and/or unnecessary.
